interface.go

```diff
@@ -32,3 +39,9 @@ type HttpServer interface {
 	ListenAndServe() error
 	Close() error
 }
+
+// AvroMarshaller defines what an avro marshaller should do
+type AvroMarshaller interface {
```

Comment:
`CovalentEncoder` might be a better naming

Reply:
No, it has nothing to do with covalent. Avro is a standalone schema with its marshaller/unmarshaller, just like json/proto